source/agora/common/EnrollmentManager.d

```diff
@@ -205,6 +205,68 @@ public class EnrollmentManager
         this.db.execute("DELETE FROM validator_set WHERE key = ?", enroll_hash[]);
     }
 
+    /***************************************************************************
+
+        In validatorSet DB, return the enrolled block height.
+
+        Params:
+            enroll_hash = key for an enrollment block height
+
+        Returns:
+            the enrolled block height.
+
+    ***************************************************************************/
+
+    public size_t getEnrolledHeight (const ref Hash enroll_hash) @safe
+    {
+        return () @trusted {
+            return this.db.execute(
+                "SELECT enrolled_height FROM validator_set WHERE key = ?",
+                enroll_hash[]).oneValue!size_t;
+        }();
+    }
+
+    /***************************************************************************
+
+        Update the enrolled height of the validatorSet DB.
+
+        Params:
+            enroll_hash = enrollment blockheight to update enroll hash
+            block_height = enrolled blockheight
+
+        Returns:
+            Update execution status
+
+    ***************************************************************************/
+
+    public bool updateEnrolledHeight (const ref Hash enroll_hash,
```

Comment:
But why is this a public method? Shouldn't the block height be provided in the `addEnrollment` method, and then it could call this `updateEnrolledHeight` method internally?

Reply:
When `updateEnrolledHeight` method is executed, it is allowed to confirm the enrolled BlockHeight at the point stored in the block after `addEnrollment` method.